e2e/cypress/integration/enterprise/cloud/billing/banner_trial_spec.js

```diff
@@ -0,0 +1,117 @@
+                contains('span', 'Town Square').should('be.visible');
+
+            cy.get('.announcement-bar__text').should('exist').within(() => {
+                // * Check for 'Your trial has started!' in banner message
+                cy.contains('span', 'days left on your Cloud trial').should('be.visible');
+            });
+
+            // # Click 'Subscribe now' button in banner message
+            cy.contains('span', 'Subscribe Now').parent().should('be.enabled').click();
+
+            // * Check for existence of 'Provide Your Payment Details' label
+            cy.contains('span', 'Provide Your Payment Details').should('be.visible');
+
+            //# Click close button in Subscribe modal
+            cy.get('#closeIcon').click();
+
+            // * Check for 'Your trial has started!' in banner message
+            cy.contains('span', 'days left on your Cloud trial').should('be.visible');
+
+            // # Click close mark
+            cy.get('.announcement-bar__close').click();
+
+            // * Check for non existence of trail banner
+            cy.contains('span', 'days left on your Cloud trial').should('not.exist');
+
+            // # Click on 'Channels' label
+            cy.contains('h1', 'Channels').should('exist').click();
+
+            // * Check for existence of trail banner in Global Header
+            cy.contains('span', 'your Cloud trial').should('be.visible');
+
+            // # Click 'Subscribe Now' button in Global Header
+            cy.contains('button', 'Subscribe Now').should('be.visible').click();
+
+            // * Check for existence of 'Provide Your Payment Details' label
+            cy.contains('span', 'Provide Your Payment Details').should('be.visible');
+
+            //# Click close button in Subscribe modal
+            cy.get('#closeIcon').click();
+
+            // * Check for 'Your trial has started!' in banner message
+            cy.contains('span', 'days left on your Cloud trial').should('be.visible');
+            cy.uiLogout();
+        });
+
+        it('MM-T4421 Non existence of Cloud trial prompts for non admin login', () => {
+            let testTeam;
+            let testUser;
+
+            // # Login as sysadmin
+            cy.apiAdminLogin();
+
+            // # Login as test user and visit town-square
+            cy.apiInitSetup().then(({team, user}) => {
+                testTeam = team;
+                testUser = user;
+            });
+
+            // # Create a new user
+            cy.apiCreateUser({prefix: 'nonAdminUser'}).then(({user}) => {
+                // # Add user to the team
+                cy.apiAddUserToTeam(testTeam.id, user.id).then(() => {
+                    // # Login as new team admin
+                    cy.apiLogin(testUser);
+
+                    // # Go to town square
+                    cy.visit(`/${testTeam.name}/channels/town-square`);
+                    cy.waitForResources();
```

Comment:
same question here, is this some kind of way to wait for DOM elements to be/not be available?